cpp/test/CMakeLists.txt

```diff
@@ -13,26 +13,39 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #=============================================================================
+enable_testing()
 
-function(ConfigureTest)
-
-  set(options OPTIONAL CUMLPRIMS MPI ML_INCLUDE RAFT_DISTRIBUTED)
-  set(oneValueArgs PREFIX NAME PATH)
-  set(multiValueArgs TARGETS CONFIGURATIONS)
+# We use rapids-cmake testing infrastructure to allow us to run multiple
+# GPU tests concurrently without causing OOM issues.
+# Use the `GPUS` and `PERCENT` options to control how 'much' of the GPUs
+# you need for a test:
+#
+# GPUS 1 PERCENT 25 -> I need 25% of a single GPU
+# GPUS 1 PERCENT 100 -> all of 1 GPU
+# GPUS 2 PERCENT 200 -> all of 2 GPUs (will only run this test on 2 GPU machines)
+include(rapids-test)
+rapids_test_init()
 
-  cmake_parse_arguments(ConfigureTest "${options}" "${oneValueArgs}"
-                          "${multiValueArgs}" ${ARGN} )
+function(ConfigureTest)
 
-  set(TEST_NAME ${ConfigureTest_PREFIX}_${ConfigureTest_NAME})
-  if(ConfigureTest_PREFIX STREQUAL "PRIMS")
-    set(TEST_INSTALL_PATH bin/gtests/libcuml_prims)
-  else()
-    set(TEST_INSTALL_PATH bin/gtests/libcuml)
+  set(options CUMLPRIMS MPI ML_INCLUDE RAFT_DISTRIBUTED)
```

Comment:
Could we get infer `ML_INCLUDE` from `PREFIX == PRIMS`? We already do that for the install component it seems, and that seems like the only case we don't want the ml includes.